### PR TITLE
Use resource discovery endpoint to determine namespaced kinds

### PR DESCRIPTION
**Pull request by adammw**
Use resource discovery endpoint to determine namespaced kinds, allowing namespaceless Custom Resources to work correctly (that currently return 404s since they are not known).

### References
https://zendesk.atlassian.net/browse/PAAS-5276

### Risks
- Low: could break existing deployments expecting previous namespace behaviour

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 1]
```diff
@@ -85,6 +85,20 @@ def self.dig_path(path)
 
     private
 
+    def namespaced_kind?(kind)
+      cluster = @doc.deploy_group.kubernetes_cluster
+      api_version = template.fetch(:apiVersion)
+
+      resources = Rails.cache.fetch(["template-filler-resources", api_version, cluster]) do
```
Comment (uthark) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]: how often cache is refreshed?
I.e. how to handle CRD updates?

**Review by uthark — APPROVED** [review]

**Comment by uthark** [comment]
🏅

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb (hunk 1 above)
Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/template_filler.rb]: apparently once per day per cluster per api version. should be "good enough"

**Review by grosser — APPROVED** [review]

**Comment by grosser** [comment]
FYI added another validation that users don't set namespace when kind does not support namespaces

**Pull request merged**